src/NumericalAlgorithms/Interpolation/PredictedZeroCrossing.cpp

```diff
@@ -16,4 +17,31 @@ double predicted_zero_crossing_value(const std::vector<double>& x_values,
   return -coefficients[0]/coefficients[1];
 }
 
+DataVector predicted_zero_crossing_value(
+    const std::deque<double>& x_values,
+    const std::deque<DataVector>& y_values) {
+  ASSERT(x_values.size() == y_values[0].size(),
+         "The x_values and y_values must be of the same size");
+  intrp::LinearLeastSquares<1> predictor{x_values.size()};
+
+  DataVector result(y_values.size());
+
+  DataVector tmp_x_values{x_values.size()};
+  for (size_t i = 0; i < x_values.size(); i++) {
+    tmp_x_values[i] = x_values[i];
+  }
```

Comment:
Why do you need tmp_x_values at all?    Does it work if you make tmp_y_values a `deque` , and then you pass two `deque`s into fit_coefficients?   (This is what I thought you were going to do; this is why I thought you added a deque to the instantiations in LinearLeastSquares.cpp)

Reply:
I don't, problem fixed, thank you